ide/spi.debugger.ui/src/org/netbeans/spi/debugger/ui/DebuggingView.java

```diff
@@ -326,6 +338,37 @@ public final void removePropertyChangeListener(PropertyChangeListener pcl) {
          */
         public void makeCurrent();
 
+        /**
+         * Get frame count of this thread. The frame count is provided when the thread
+         * is suspended only.
+         * @return the frame count, <code>0</code> when the thread is running.
+         */
+        public default int getFrameCount() {
+            return 0;
+        }
+
+        /**
+         * Get the stack frames of this thread. Stack frames are provided when the thread
+         * is suspended only.
+         *
+         * @return a list of stack frames, it's empty when the thread is running.
+         */
+        public default List<DVFrame> getFrames() {
```

Comment:
Recently I wrote a [philippic against using default methods](http://wiki.apidesign.org/wiki/BinaryCompatibleDefaultMethods), but given the [one to many proximity](http://wiki.apidesign.org/wiki/Proximity) - e.g. there is very likely just a single implementation of these interfaces in the debugger impl modules - I expect no clashes and `default` methods are the easiest way to extend the interfaces.

Reply:
Right.